integrations/firehydrant/client.py

```diff
@@ -0,0 +1,451 @@
+
+        # Calculate the duration if both "started" and "resolved" events are found.
+        if started_time and resolved_time:
+            duration = resolved_time - started_time
+            duration_hours = duration.total_seconds() / 60  # Convert to minutes
+            return round(duration_hours, 2)
+
+        return None  # Return None if either "started" or "resolved" event is missing.
+
+    async def get_retrospective_questions_and_answers(
+        self, questions: list[dict[str, Any]]
+    ) -> list[dict[str, Any]]:
+        """Format retrospective questions by renaming title to 'question' and body to 'answer'.
+
+        Args:
+            questions (list[dict[str, Any]]): A list of retrospective question objects.
+
+        Returns:
+            list[dict[str, Any]]: A list of formatted retrospective questions with 'question' and 'answer' keys.
+        """
+
+        retro_responses_list = []
+        for question in questions:
+            retro_responses_list.append(
+                {"question": question.get("title"), "answer": question.get("body")}
+            )
+        return retro_responses_list
+
+    async def compute_service_mean_time_metrics(
+        self, active_incidents_ids: list[Any]
+    ) -> dict[str, Any]:
+        """Computes mean time service details for active incidents.
+
+        Args:
+            active_incidents_ids (list[Any]): A list of active incident IDs.
+
+        Returns:
+            dict[str, Any]: A dictionary containing computed mean time service metrics,
+            including time to acknowledge, time to identify, time to mitigate, and time to resolve.
+
+        Note:
+            This function calculates various incident analytics metrics based on milestone data.
+            The metrics may include Mean Time to Detection (MTTD), Mean Time to Acknowledged (MTTA),
+            Mean Time to Mitigation (MTTM), and Mean Time to Resolution (MTTR).
+        """
+
+        if len(active_incidents_ids) == 0:
+            return {}
+
+        service_metrics = {
+            "time_to_acknowledge": 0,
+            "time_to_identify": 0,
+            "time_to_mitigate": 0,
+            "time_to_resolve": 0,
+        }
+
+        for incident_id in active_incidents_ids:
+            incident_data = await self.get_single_incident(incident_id=incident_id)
+            milestones = incident_data.get("milestones", {})
+            incident_metrics = await self.compute_incident_analytics_metrics(
+                milestones=milestones
+            )
+
+            # Update the cumulative metrics for the service
+            for metric_name, metric_value in incident_metrics.items():
+                service_metrics[metric_name] += metric_value
+
+        # Calculate the average metrics for the service
+        for metric_name in service_metrics:
+            service_metrics[metric_name] //= len(active_incidents_ids)
+
+        return service_metrics
```

Comment:
correct me if im worng but this can be done using jq/ calculation property

Reply:
I tried to move this to jq but it wasn't successful